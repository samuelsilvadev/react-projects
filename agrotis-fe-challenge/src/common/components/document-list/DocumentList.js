import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import { gteSmallMedia } from '../../media.style';

const Ul = styled.ul`
	list-style: none;
	padding: 2rem;
`;

const Li = styled.li`
	border: 0.1rem solid #ccc;
	border-radius: 0.5rem;
	display: flex;
	flex-direction: column;
	padding: 2rem;

	&:not(:last-of-type) {
		margin-bottom: 1.5rem;
	}

	${gteSmallMedia} {
		flex-direction: row;
	}
`;

const Div = styled.div`
	width: ${({ width }) => width};
`;

const Span = styled.span`
	color: #9f9f9f;
`;

const StrongSpan = styled.span`
	display: inline-block;
	font-weight: bold;
`;

function DocumentList(props) {
	const { documents = [] } = props;

	const renderListItem = (document, key) => {
		const { name, description } = document;

		return (
			<Li key={key}>
				<Div width="35rem">
					<Span>Name: </Span>
					<StrongSpan>{name}</StrongSpan>
				</Div>
				<Div>
					<Span>Description: </Span>
					<StrongSpan>{description}</StrongSpan>
				</Div>
			</Li>
		);
	};

	return <Ul>{documents.map(renderListItem)}</Ul>;
}

DocumentList.propTypes = {
	documents: PropTypes.arrayOf(PropTypes.object)
};

export default DocumentList;
