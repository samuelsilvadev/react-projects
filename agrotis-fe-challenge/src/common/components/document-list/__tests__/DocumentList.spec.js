import React from 'react';
import { render } from '@testing-library/react';

import DocumentList from '../DocumentList';

describe('<DocumentList />', () => {
	it('should render correctly when has `documents` prop', () => {
		const documents = [
			{
				name: 'Document 1',
				description: 'Description 1'
			}
		];
		const { container, queryByText } = render(
			<DocumentList documents={documents} />
		);

		expect(queryByText(documents[0].name)).toBeVisible();
		expect(queryByText(documents[0].description)).toBeVisible();
		expect(container).toMatchSnapshot();
	});

	it('should not render if has no `documents` prop', () => {
		const { container, queryByTestId } = render(<DocumentList />);

		expect(queryByTestId('documentList')).not.toBeInTheDocument();
		expect(container).toMatchSnapshot();
	});
});
