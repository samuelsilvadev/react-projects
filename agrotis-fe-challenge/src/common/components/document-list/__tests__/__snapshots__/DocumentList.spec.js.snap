// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DocumentList /> should not render if has no \`documents\` prop 1`] = `<div />`;

exports[`<DocumentList /> should render correctly when has \`documents\` prop 1`] = `
<div>
  <ul
    class="sc-bdVaJa gNTQhD"
    data-test="documentList"
  >
    <li
      class="sc-bwzfXH fyZOft"
    >
      <div
        class="sc-htpNat gkdNxb"
        width="35rem"
      >
        <span
          class="sc-bxivhb csUzPq"
        >
          Name: 
        </span>
        <span
          class="sc-ifAKCX fnqLfI"
        >
          Document 1
        </span>
      </div>
      <div
        class="sc-htpNat gsmuIY"
      >
        <span
          class="sc-bxivhb csUzPq"
        >
          Description: 
        </span>
        <span
          class="sc-ifAKCX fnqLfI"
        >
          Description 1
        </span>
      </div>
    </li>
  </ul>
</div>
`;
